ts/features/idpay/unsubscription/xstate/services.ts

```diff
@@ -0,0 +1,88 @@
+/* eslint-disable no-underscore-dangle */
+import * as E from "fp-ts/lib/Either";
+import * as TE from "fp-ts/lib/TaskEither";
+import { flow, pipe } from "fp-ts/lib/function";
+import { PreferredLanguage } from "../../../../../definitions/backend/PreferredLanguage";
+import { InitiativeDTO } from "../../../../../definitions/idpay/InitiativeDTO";
+import { IDPayClient } from "../../common/api/client";
+import { Context } from "./context";
+
+export type Services = {
+  getInitiativeInfo: {
+    data: InitiativeDTO;
+  };
+  unsubscribeFromInitiative: {
+    data: undefined;
+  };
+};
+
+const createServicesImplementation = (
+  client: IDPayClient,
+  token: string,
+  language: PreferredLanguage
+) => {
+  const getInitiativeInfo = async (
+    context: Context
+  ): Promise<InitiativeDTO> => {
+    const dataResponse = await TE.tryCatch(
+      async () =>
+        await client.getWalletDetail({
+          bearerAuth: token,
+          "Accept-Language": language,
+          initiativeId: context.initiativeId
+        }),
+      E.toError
+    )();
+
+    return pipe(
+      dataResponse,
+      E.fold(
+        () => Promise.reject(undefined),
+        flow(
+          E.fold(
+            () => Promise.reject(undefined),
+            response =>
+              response.status !== 200
+                ? Promise.reject(undefined)
+                : Promise.resolve(response.value)
+          )
```

Comment:
`flow` function is like a `pipe` but instead of passing a value as first argument you can use a left-to-right composition to get a function:

A practical example:
```
pipe(1, multiply2, toString)
// is equal to
flow(multiply2, toString)(1)
```

It's useful when you want to avoid to use anonymous functions. I think it fits this case.

Reply:
```suggestion
          E.fold(
            () => Promise.reject(undefined),
            response =>
              response.status !== 200
                ? Promise.reject(undefined)
                : Promise.resolve(response.value)
```
why not just remove it though? to me it seems like you're just wrapping `E.fold` in a composer function, just to get it back. Am I missing something?

Reply:
In https://github.com/pagopa/io-app/pull/4462/commits/a84349600c2eef5e37bb8b3a9e26542e439c672e I made a refactoring which implements the best case for the `flow` operator. I think it's more readable.